### ISSUE TITLE
EOF Implementers Call #62

### ISSUE THREAD

**Issue by poojaranjan** [ethereum/pm#1192]
## Meeting Info
[Nov 27, 2024 , 15:00 UTC](https://www.timeanddate.com/worldclock/converter.html?iso=20240403T150000&p1=1440&p2=37&p3=136&p4=237&p5=923&p6=204&p7=671&p8=16&p9=41&p10=107&p11=28)

Duration: 60 minutes

Zoom: [https://us02web.zoom.us/j/88940506383?pwd=aTdsbHVyMTNDSUFHYmhTWlI2ZEVldz09](https://www.google.com/url?q=https://us02web.zoom.us/j/88940506383?pwd%3DaTdsbHVyMTNDSUFHYmhTWlI2ZEVldz09&sa=D&source=calendar&usd=2&usg=AOvVaw0oj74WBiSnw1UTI1ussX15)

📅 Subscribe to the [Ethereum Protocol Call calendar](https://calendar.google.com/calendar/u/1?cid=Y191cGFvZm9uZzhtZ3JtcmtlZ243aWM3aGs1c0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t) for calendar invites

### Resources
- [Current specs](https://github.com/ipsilon/eof/blob/main/spec/eof.md)
    - [EIP-7692: EVM Object Format (EOFv1) Meta](https://eips.ethereum.org/EIPS/eip-7692)
    - EIPs Included <details>
[EIP-3540](https://eips.ethereum.org/EIPS/eip-3540): EOF - EVM Object Format v1
[EIP-3670](https://eips.ethereum.org/EIPS/eip-3670): EOF - Code Validation
[EIP-4200](https://eips.ethereum.org/EIPS/eip-4200): EOF - Static relative jumps
[EIP-4750](https://eips.ethereum.org/EIPS/eip-4750): EOF - Functions
[EIP-5450](https://eips.ethereum.org/EIPS/eip-5450): EOF - Stack Validation
[EIP-6206](https://eips.ethereum.org/EIPS/eip-6206): EOF - JUMPF and non-returning functions
[EIP-7480](https://eips.ethereum.org/EIPS/eip-7480): EOF - Data section access instructions
[EIP-663](https://eips.ethereum.org/EIPS/eip-663): SWAPN, DUPN and EXCHANGE instructions
[EIP-7069](https://eips.ethereum.org/EIPS/eip-7069): Revamped CALL instructions
[EIP-7620](https://eips.ethereum.org/EIPS/eip-7620): EOF Contract Creation
[EIP-7698](https://eips.ethereum.org/EIPS/eip-7698): EOF - Creation transaction </details>
- [EOF Meeting Playlist](https://www.youtube.com/playlist?list=PLJqWcTqh_zKHu7Z91YVYWy_PSmOc_66-F) 
- EVM (EOF) [Taxonomy](https://rakita.github.io/blog/EVM%20(EOF)%20Taxonomy.png)
- EOF explainers on [PEEPanEIP](https://www.youtube.com/playlist?list=PL4cwHXAawZxqOHV_F40AJbzcl8b6tG8xw) <details>
[EVM Object Format (EOF) with Danno Ferrin](https://youtu.be/rlZwPuF149U)
[EVM Object Format (EIP-3540 & EIP-3670) with Andrei Maiboroda, Axic & Pawel](https://youtu.be/GMeRA-xPp-E) </details>
- [EOF Testing Coverage Tracker](https://github.com/ethereum/execution-spec-tests/blob/main/tests/prague/eip7692_eof_v1/tracker.md)
  - [Implementation Matrix](https://github.com/ipsilon/eof/blob/main/spec/implementation_matrix.md)
- [Solidity + EOF POC summary](https://notes.ethereum.org/@ipsilon/solidity_eof_poc)
- Matrix on [evmobjectformat.org](https://evmobjectformat.org/)
- Prev meeting [notes](https://github.com/ethereum/pm/issues/1184#issuecomment-2447633714)

## Agenda
- Testing updates 
- Client updates
- Compiler updates
- Spec Updates 
- Other
  - Separate contract metadata data section Ref: [here](https://github.com/ethereum/pm/issues/1192#issuecomment-2488656792)

 
Please add other agenda items or links to discuss.

### Next call on Dec 11, 2024

**Comment by kuzdogan** [comment]
Proposing the agenda item "separate contract metadata data section" 

TLDR: We believe it'd make sense to have a data section for contract metadata: data that won't be used by and is unreachable to the contract's code. 

Our main use case is the [Solidity's contract metadata hash](https://docs.soliditylang.org/en/latest/metadata.html#encoding-of-the-metadata-hash-in-the-bytecode) + the Solidity version, which are appended to the bytecode in CBOR encoding (See it in action at [playground.sourcify.dev](https://playground.sourcify.dev/)). This wouldn't be the only use case as other languages and tools might want to add various metadata to the contract's bytecode. 

With its current form putting any metadata in the data section would cause changes in the contract's code because of shifting offsets. Any change in the contract metadata should not cause changes in the code. This is particularly important for source code verification, which is sensitive to changes in the bytecode. Having a designated place for the metadata would also make things a lot easier for verifiers, as this data has to be omitted during verification and it's difficult to find where it is in the bytecode (See [the blog post](https://docs.sourcify.dev/blog/finding-auxdatas-in-bytecode/) for more info)

**Comment by pdobacz** [comment]
I'd like to discuss an alternative proposed EOFCREATE hashing scheme https://github.com/ipsilon/eof/issues/162#issuecomment-2491696792

Also pls reopen, this call is scheduled for today

**Comment by frangio** [comment]
I'd also like to discuss generic factory contracts.

**Comment by shemnon** [comment]
Here's the list of potential Pectra-0 upgrades, for reference.  I don't think we need to re-hash any of it until osaka-0 launces

  * EXT*CALL return codes
  * EOFCREATE Hashing:  https://github.com/ipsilon/eof/issues/162
  * https://github.com/ipsilon/eof/issues/134
  * https://github.com/ipsilon/eof/pull/156
  * https://github.com/ipsilon/eof/pull/164 rename RETURNCONTRACT
  * Don't forget last weeks discussion, but we won't be moving on those until after devcon
    * HASCODE/EXTCODESIZE

**Comment by shemnon** [comment]
One idea for a tempurature check: change the data section number to something higher like 0x7f or 0xff, so it will always sort near or at the end of section numbers.

**Comment by shemnon** [comment]
Also, state tests with invalid EOF, what to do?

**Comment by shemnon** [comment]
Testing Update
* How to handle State tests with invalid EOF?
  * state tests - reject test if any EOF is invalid
  * Block tests - only an issue in genesis?  Abort if EOF in genesis is invalid.
  * Imported blocks - presume valid as create TXes are how they are added, so invalid EOF should result in an failed transactions.
  * Extends to 7702 - 0xEF01 validation?
* EOFWrap Tests
  * ports over legacy tests into EOF if it ports, stopgap for full testing
  * https://github.com/ethereum/execution-spec-tests/issues/972 will ultimately port all old tests
 
Client and Compiler Updates
* No client Updates, mostly focused on petctra
* Solidity working on EOF as an experimental feature 
  * https://github.com/ethereum/solidity/pull/15550
  * https://github.com/ethereum/solidity/pull/15555
  * https://github.com/ethereum/solidity/pull/15559

Spec Updates
* Compiler Metadata Section
  * Kaan from Sourcify Team
  * Current practice is to just append
  * would want a separate metadata section in EOF. 
    * Unreachable by code (a good thing)
    * contains the CBOR data solidity produces
  * Current status of appended to data and behind constructor fields makes it hard to find
  * Experimental Solidity EOF handling is to put CBOR metadata at the beginning of the data section.
  * Would insulate code/data indexes from variable CBOR sizes, such as if experimental flags are logged.
  * Next step is an EIP

Brief discussion on header section numbers

EOFCREATE hash - https://github.com/ipsilon/eof/issues/162#issuecomment-2491696792
  * danno wants a "0xef01" hash added
  * Solidity has concerns about the genericness, would prefer container index
  * Bad salt management could prohibit multiple deployments
  * Should hash include auxdata, not just code data?
  * possible issues with cross-chain deployment.  The more mandatory data makes same address contracts difficult.
  * note: some people don't like metadata has in CREATE2, would compiler metadata be excluded from address derivation?
  * Are there security implications? Would the "code hash" guarantees be forgotten about?  Could compilers compensate?
  * Please add comments to the thread.

**Comment by poojaranjan** [comment]
Recording: https://youtu.be/yzYUWpa-1QM

**Comment by poojaranjan** [comment]
Closed in favor of #1205
